src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -704,43 +718,77 @@ void sys_network_bsd(int queue_fd, const char* LOCATION){
     os_calloc(i, sizeof(char *), ifaces_list);
 
     /* Create interfaces list */
-    for (ifa = ifaddrs_ptr; ifa; ifa = ifa->ifa_next){
-        found = 0;
-        for (i=0; i<=j; i++){
-            if (!ifaces_list[i]){
-                if (ifa->ifa_flags & IFF_LOOPBACK)
-                    found = 1;
-
-                break;
-
-            }else if (!strcmp(ifaces_list[i], ifa->ifa_name)){
-                    found = 1;
-                    break;
-            }
-        }
-        if (!found){
-
-            ifaces_list[j] = strdup(ifa->ifa_name);
-            j++;
-        }
-    }
+    size_ifaces = getIfaceslist(ifaces_list, ifaddrs_ptr);
 
-    if(!ifaces_list[j-1]){
+    if(!ifaces_list[size_ifaces-1]){
         mterror(WM_SYS_LOGTAG, "Not found any interface. Network inventory suspended.");
         return;
     }
 
-    for (i=0; i<j; i++){
+    #if defined(__MACH__)
+    gateways = OSHash_Create();
+    if (getGatewayList(gateways) < 0){
+        merror("Error creating the list of gateways");
```

Comment:
Should use `mterror(WM_SYS_LOGTAG, "...`. Review the messages as well: `Unable to obtain the Default Gateway list`